subsys/bluetooth/mesh/access.c

```diff
@@ -342,8 +350,6 @@ void bt_mesh_comp_unprovision(void)
 	BT_DBG("");
 
 	dev_primary_addr = BT_MESH_ADDR_UNASSIGNED;
-
-	bt_mesh_model_foreach(mod_init, NULL);
```

Comment:
I'm wondering if we should do here the reverse of what comp_provision() does, i.e. iterate all elements and set their addresses to UNASSIGNED as well?

Reply:
That said, I guess the code wasn't doing it so far (only making the per-model calls), so that can be left as an independent (later) fix

Reply:
Good point, yes. I think there are several parts of the unprovision procedure that are a bit fuzzy, and I'm not sure I improved it by doing this. 

After removing this call, we're no longer stopping periodic publishing, but before, we would reinitialize publish's k_delayed_work, which breaks the global timeout queue by wiping the timeout dlist node.

Reply:
Seems like something to be fixed in the config server’s reset callback? Anyway, I guess we should hold off on merging this if it’s causing regressions (even though the current code isn’t completely bug-free either)

Reply:
@trond-snekvik 

> Good point, yes. I think there are several parts of the unprovision procedure that are a bit fuzzy, and I'm not sure I improved it by doing this.
> 
> After removing this call, we're no longer stopping periodic publishing, but before, we would reinitialize publish's k_delayed_work, which breaks the global timeout queue by wiping the timeout dlist node.

Sure, reinitialize timer indeed is a problem, it's really an unexpected discovery. 

but, what mean `we're no longer stopping periodic publishing` ??? , i'm find k_delayed_work already canceled in `b_mesh_reset -> bt_mesh_subnet_del -> bt_mesh_app_key_del -> mod_unbind `func.

Reply:
> I'm wondering if we should do here the reverse of what comp_provision() does, i.e. iterate all elements and set their addresses to UNASSIGNED as well?

I agree, it can be necessary to do this in another commit.

Reply:
> i'm find k_delayed_work already canceled in `b_mesh_reset -> bt_mesh_subnet_del -> bt_mesh_app_key_del -> mod_unbind` func.

Well spotted, you're right! I missed that.

So if this isn't a regression after all, I think it could be safe to merge this after the release.